pkg/controllers/user/pipeline/controller/clusterpipeline/clusterpipelinesyncer.go

```diff
@@ -0,0 +1,108 @@
+package clusterpipeline
+
+import (
+	"github.com/pkg/errors"
+	"github.com/rancher/rancher/pkg/controllers/user/pipeline/utils"
+	"github.com/rancher/types/apis/core/v1"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type Syncer struct {
+	cluster    *config.UserContext
+	namespaces v1.NamespaceInterface
+}
+
+func Register(cluster *config.UserContext) {
+	clusterPipelines := cluster.Management.Management.ClusterPipelines("")
+	clusterPipelineSyncer := &Syncer{
+		cluster: cluster,
+	}
+	clusterPipelines.AddHandler("cluster-pipeline-syncer", clusterPipelineSyncer.Sync)
```

Comment:
Per my understanding, controller in user package is created per cluster, which means we should use AddClusteredScopedHandler, otherwise, we will do sync for obj of other clusters, which is unnessesary.

Reply:
@alena1108  am i correct?

Reply:
@zionwu correct